//
// Copyright 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
// These materials are licensed under the Amazon Software License in connection with the Alexa Gadgets Program.
// The Agreement is available at https://aws.amazon.com/asl/.
// See the Agreement for the specific terms and conditions of the Agreement.
// Capitalized terms not defined in this file have the meanings given to them in the Agreement.
//

syntax = "proto3";

import "common.proto";
import "device.proto";
import "firmware.proto";

enum Command {
    NONE = 0;
    GET_DEVICE_INFORMATION = 20;
    GET_DEVICE_FEATURES = 28;
    UPDATE_COMPONENT_SEGMENT = 94;
    APPLY_FIRMWARE = 95;
}

message Response {
    ErrorCode error_code = 1;
    oneof payload {
        DeviceInformation device_information = 3;
        DeviceFeatures device_features = 28;
        FirmwareComponent firmware_component = 2;
    }
}

message ControlEnvelope {
    Command command = 1;
    oneof payload {
        Response response = 9;
        GetDeviceInformation get_device_information = 20;
        GetDeviceFeatures get_device_features = 28;
        UpdateComponentSegment update_component_segment = 94;
        ApplyFirmware apply_firmware = 95;
    }
}
